src/mmds/src/data_store.rs

```diff
@@ -112,21 +123,53 @@ impl Mmds {
     }
 
     /// Set the MMDS version.
-    pub fn set_version(&mut self, version: MmdsVersion) {
-        self.version = version;
+    pub fn set_version(&mut self, version: MmdsVersion) -> Result<(), Error> {
+        match version {
+            MmdsVersion::V1 => {
+                self.token_authority = None;
+                Ok(())
+            }
+            MmdsVersion::V2 => {
+                if self.token_authority.is_none() {
+                    self.token_authority =
+                        Some(TokenAuthority::new().map_err(Error::TokenAuthority)?);
+                }
+                Ok(())
+            }
+        }
     }
 
-    /// Return the MMDS version.
+    /// Return the MMDS version by checking the token authority field.
     pub fn version(&self) -> MmdsVersion {
-        self.version
+        if self.token_authority.is_none() {
+            MmdsVersion::V1
+        } else {
+            MmdsVersion::V2
+        }
+    }
+
+    /// Checks if the provided token has not expired.
+    pub fn is_valid_token(&self, token: &str) -> bool {
+        match &self.token_authority {
+            Some(ta) => ta.is_valid(token),
+            None => unreachable!(),
```

Comment:
I missed this in the previous reviews. We should not have `unreachable!()` statements in the public interface. All public functions must offer panic-free functionality. I saw that the code logic will never actually call this without a token authority, but we should make the check at a higher level and offer an error here instead (like `InvalidState` or something like that). Another option would be to make this function private and create a safe public wrapper, but it feels clunky since it's not an internal function and it just calls `TokenAuthority::is_valid()`.